gnd/src/main/java/com/google/android/gnd/repository/FeatureRepository.java

```diff
@@ -127,17 +128,18 @@ private Completable updateLocalFeature(RemoteDataEvent<Feature> event) {
   }
 
   private FeatureMutation fromFeature(Feature feature, Type type) {
-    Optional<Point> newLocation =
-        feature instanceof PointFeature
-            ? Optional.of(((PointFeature) feature).getPoint())
-            : Optional.empty();
     return FeatureMutation.builder()
         .setType(type)
         .setSyncStatus(SyncStatus.PENDING)
         .setProjectId(feature.getProject().getId())
         .setFeatureId(feature.getId())
         .setLayerId(feature.getLayer().getId())
-        .setNewLocation(newLocation)
+        .setNewLocation(
```

Comment:
Instead of setting both location and vertices, please refactor this so that only the relevant field is set based on the type of feature.

Reply:
updated